### PR TITLE
Feature 53859 security audit smaller issues

### PR DESCRIPTION
**Pull request by LBruegger**

### CODE REVIEW

File: app/assets/javascripts/modules/login.js.coffee [hunk 1]
```diff
@@ -28,7 +28,7 @@ class app.Login
     return
 
   setLocalUsername = ->
```
Comment (psunix) [inline app/assets/javascripts/modules/login.js.coffee:30]: remove function completely !

File: app/controllers/api_controller.rb [hunk 2]
```diff
@@ -88,6 +89,12 @@ def authorize_with_headers
     end
   end
 
+  def set_headers
+    response.headers['Content-Type'] = 'text/json'
+    response.headers['Content-disposition'] = "attachment;\
+    filename='" + params['controller'].to_s + "'.json"
```
Comment (psunix) [inline app/controllers/api_controller.rb:95]: try to avoid using params for static values.
I would rather use request.env['PATH_INFO']

Comment (psunix) [reply app/controllers/api_controller.rb:95]: add this header checking at least to one controller spec

File: config/environments/production.rb [hunk 3]
```diff
@@ -96,4 +96,6 @@
 
   # Do not dump schema after migrations.
   config.active_record.dump_schema_after_migration = false
+
+  config.force_ssl = true
```
Comment (psunix) [inline config/environments/production.rb:100]: remove !
it's already in this file -> line 53

File: config/initializers/session_store.rb [hunk 4]
```diff
@@ -1,9 +1,8 @@
 # Be sure to restart your server when you modify this file.
 
-Rails.application.config.session_store :cache_store, key: '_cryptopus_session', expire_after: 5.minutes
+Rails.application.config.session_store :cache_store, key: '_cryptopus_session', expire_after: 5.minutes, same_site: :lax
```
Comment (psunix) [inline config/initializers/session_store.rb:3]: same_site: :strict doesn't work, does it ?

**Comment by psunix** [comment]
maybe you could add an integration test named http_headers_spec.rb with a section 'security'
which checks all those headers.

File: config/initializers/session_store.rb (hunk 4 above)
Comment (LBruegger) [reply config/initializers/session_store.rb]: It truly does work 😅 yesterday, it didn't

**Pull request merged**